Chummer/Classes/clsCustomDataDirectoryInfo.cs

```diff
@@ -0,0 +1,611 @@
+
+        /// <summary>
+        /// A list of all dependencies each formatted as Tuple(guid, str name, int version, bool isMinimumVersion).
+        /// </summary>
+        public IReadOnlyList<DirectoryDependency> DependenciesList => _lstDependencies;
+
+        /// <summary>
+        /// A list of all exclusivities each formatted as Tuple(int guid, str name, int version, bool ignoreVersion).
+        /// </summary>
+        public IReadOnlyList<DirectoryDependency> ExclusivitiesList => _lstExclusivities;
+
+        /// <summary>
+        /// A Dictionary containing all Descriptions, which uses the language code as key
+        /// </summary>
+        public IReadOnlyDictionary<string, string> DescriptionDictionary => _descriptionDictionary;
+
+        /// <summary>
+        /// The description to display
+        /// </summary>
+        public string DisplayDescription { get; private set; }
+
+        /// <summary>
+        /// A Dictionary, that uses the author name as key and provides a bool if he is the main author
+        /// </summary>
+        public IReadOnlyDictionary<string, bool> AuthorDictionary => _authorDictionary;
+
+        /// <summary>
+        /// A string containing all Authors formatted as Author(main), Author2 
+        /// </summary>
+        public string DisplayAuthors { get; private set; }
+
+        public Guid Guid => _guid;
+
+        /// <summary>
+        /// The name including the Version in this format "NAME (Version)"
+        /// </summary>
+        public string DisplayName { get; }
+        #endregion
+
+        #region Interface Implementations and Operators
+        public int CompareTo(object obj)
+        {
+            if (obj is CustomDataDirectoryInfo objOtherDirectoryInfo)
+                return CompareTo(objOtherDirectoryInfo);
+            return 1;
+        }
+
+        public int CompareTo(CustomDataDirectoryInfo other)
+        {
+            int intReturn = string.Compare(Name, other.Name, StringComparison.Ordinal);
+            if (intReturn == 0)
+                intReturn = string.Compare(Path, other.Path, StringComparison.Ordinal);
+            //Should basically never happen, because paths are supposed to be unique. But this "future proofs" it.
+            if (intReturn == 0)
+                intReturn = string.Compare(Version.ToString(GlobalOptions.CultureInfo), other.Version.ToString(GlobalOptions.CultureInfo), StringComparison.Ordinal);
+            
+            return intReturn;
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj is CustomDataDirectoryInfo objOther)
+                return Equals(objOther);
+            return false;
+        }
+
+        public override int GetHashCode()
+        {
+            //Path and Guid should already be enough because they are both unique, but just to be sure.
+            return new { Name, Path, Guid, Version, DependenciesList, ExclusivitiesList }.GetHashCode();
```

Comment:
So there is `EnumeralExtension.GetEnsembleHashCode<T>()` which produces a hash code based on the content of the `IEnumerable`. I can't however just include those in the `GetHashCode()`.

Calculating them beforehand and then including the pointer to those local variables like this:

``` c#
        public override int GetHashCode()
        {
            var dependencyHash = DependenciesList.GetEnsembleHashCode();
            var exclusivityHash = ExclusivitiesList.GetEnsembleHashCode();
            //Path and Guid should already be enough because they are both unique, but just to be sure.
            return new { Name, Path, Guid, Version, exclusivityHash, dependencyHash }.GetHashCode();
        }
```
would work.

But my problem with this is, that I'm not sure if I understood everything.
What I think this way of doing it would do compared to the previous implementation is as follows.

`GetHashCode` takes in all the stuff given in the brackets {} and calculates a Hash. But it can't use a `IEnumerable`, so there instead of the content the pointer is used to calculate the hash.
If I now give it the `Int` hashes of those lists instead of the lists themselves those integers are used for the Hash calculation and in extension our resulting hash depends on the content of those lists.